docs/content/querying/sql.md

```diff
@@ -2,6 +2,130 @@
+retrieve column metadata, use the query:
+
+```sql
+SELECT * FROM metadata.COLUMNS
+```
+
+If `druid.sql.planner.useFallback` is enabled, full SQL is possible on metadata tables. However, useFallback is not
+recommended in production since it can generate unscalable query plans. The JDBC driver allows accessing
+table and column metadata through `connection.getMetaData()` even if useFallback is off.
+
+### Time functions
+
+Druid's SQL language supports a number of time operations, including:
+
+- `FLOOR(__time TO <granularity>)` for grouping or filtering on time buckets, like `SELECT FLOOR(__time TO MONTH), SUM(cnt) FROM druid.foo GROUP BY FLOOR(__time TO MONTH)`
+- `EXTRACT(<granularity> FROM __time)` for grouping or filtering on time parts, like `SELECT EXTRACT(HOUR FROM __time), SUM(cnt) FROM druid.foo GROUP BY EXTRACT(HOUR FROM __time)`
+- Comparisons to `TIMESTAMP '<time string>'` for time filters, like `SELECT COUNT(*) FROM druid.foo WHERE __time >= TIMESTAMP '2000-01-01 00:00:00' AND __time < TIMESTAMP '2001-01-01 00:00:00'`
+
+### Semi-joins
+
+Semi-joins involving `IN (SELECT ...)`, like the following, are planned with a special process.
+
+```sql
+SELECT x, count(*)
+FROM druid.foo
+WHERE x IN (SELECT x FROM druid.bar WHERE y = 'baz')
+GROUP BY x
+```
+
+For this query, the broker will first translate the inner select on dataSource `bar` into a groupBy to find distinct
+`x` values. Then it'll use those distinct values to build an "in" filter on dataSource `foo` for the outer query. The
+configuration parameter `druid.sql.planner.maxSemiJoinRowsInMemory` controls the maximum number of values that will be
+materialized for this kind of plan.
+
+### Configuration
+
+Druid's SQL planner can be configured on the [Broker node](../configuration/broker.html#sql-planner-configuration).
+
+### Unsupported features
+
+Druid does not support all SQL features. Most of these are due to missing features in Druid's native JSON-based query
+language. Some unsupported SQL features include:
+
+- Grouping on functions of multiple columns, like concatenation: `SELECT COUNT(*) FROM druid.foo GROUP BY dim1 || ' ' || dim2`
+- Grouping on long and float columns.
+- Filtering on float columns.
+- Filtering on non-boolean interactions between columns, like two columns equaling each other: `SELECT COUNT(*) FROM druid.foo WHERE dim1 = dim2`.
+- A number of miscellaneous functions, like `TRIM`.
+- Joins, other than semi-joins as described above.
+
+Additionally, some Druid features are not supported by the SQL language. Some unsupported Druid features include:
+
+- [Multi-value dimensions](multi-value-dimensions.html).
+- [Query-time lookups](lookups.html).
+- [Nested groupBy queries](groupbyquery.html#nested-groupbys).
+- Extensions, including [approximate histograms](../development/extensions-core/approximate-histograms.html) and
+[DataSketches](../development/extensions-core/datasketches-aggregators.html).
```

Comment:
@b-slim just because HLL is in core. I haven't added an extension point yet (was hoping to do that in a future PR) and so only features in core Druid are usable right now.